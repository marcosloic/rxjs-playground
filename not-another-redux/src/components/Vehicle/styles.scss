.vehicle {
    flex: auto;
    height: 250px;
    min-width: 150px;
    max-width: 250px;
    margin: 0 8px 8px 0;
    background: #fafafa;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);

    &:hover {
        box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25),
            0 10px 10px rgba(0, 0, 0, 0.22);
    }

    img {
        //Repeating the values is a bit meh, but it lets us handle missing images with just css
        //Instead of using an onError callback
        min-height: 150px;
        max-height: 150px;
        min-width: 100%;
        max-width: 100%;
        object-fit: cover;
        background-image: url('../../assets/fallback.jpg');
        background-size: cover;
        display: block;
    }

    .content {
        display: flex;
        flex-direction: column;
        margin: 5px;
        align-items: self-end;
    }

    @media screen and (max-width: 543px) {
        max-width: 100%;
    }
}
