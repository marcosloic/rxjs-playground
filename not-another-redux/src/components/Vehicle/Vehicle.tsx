import React from 'react';
import { ApiData } from '../../types/ApiData.interface';
import './styles.scss';
import { ColorTag } from '../ColorTag/ColorTag';

type VehiclePropTypes = {
    details: ApiData;
};

export const Vehicle: React.FunctionComponent<VehiclePropTypes> = React.memo(
    props => {
        const { details } = props;
        const { brand, type, colors, img } = details;
        return (
            <article className={'vehicle'}>
                <img src={img} alt={brand} />
                <div className={'content'}>
                    <p>
                        <b>Brand:</b> {brand}
                    </p>
                    <p>
                        <b>Type:</b> {type}
                    </p>
                    <p>
                        <b>Available in:</b>{' '}
                        {colors &&
                            colors
                                .sort()
                                .map(color => (
                                    <ColorTag key={color} color={color} />
                                ))}
                    </p>
                </div>
            </article>
        );
    }
);

Vehicle.displayName = 'Vehicle';
