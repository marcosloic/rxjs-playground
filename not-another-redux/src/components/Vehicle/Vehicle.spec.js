import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Vehicle } from './Vehicle';

const fakeDetails = {
    brand: 'brand',
    type: 'type',
    colors: ['colors'],
    img: 'img'
};

describe('The Vehicle component', () => {
    it('Matches the snapshot', () => {
        const vehicle = TestRenderer.create(<Vehicle details={fakeDetails} />);
        expect(vehicle.toJSON()).toMatchSnapshot();
    });

    it('renders the vehicle details', () => {
        const vehicle = TestRenderer.create(
            <Vehicle details={fakeDetails} />
        ).toJSON();
        const [img, div] = vehicle.children;
        expect(vehicle.children.length).toBe(2);
        expect(img.type).toBe('img');
        expect(div.children.length).toBe(3);
    });
});
