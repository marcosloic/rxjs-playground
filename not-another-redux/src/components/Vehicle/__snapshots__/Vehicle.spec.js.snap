// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Vehicle component Matches the snapshot 1`] = `
<article
  className="vehicle"
>
  <img
    alt="brand"
    src="img"
  />
  <div
    className="content"
  >
    <p>
      <b>
        Brand:
      </b>
       
      brand
    </p>
    <p>
      <b>
        Type:
      </b>
       
      type
    </p>
    <p>
      <b>
        Available in:
      </b>
       
      <span
        className="colorTag"
        style={
          Object {
            "backgroundColor": "colors",
          }
        }
      />
    </p>
  </div>
</article>
`;
